### PR TITLE
Prepare 0.6.0 release

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
cc @rohitgr7 

<details><summary>commits since 0.5.0</summary>

```
c2552c7 (origin/master, origin/HEAD, master) set min PL as 1.7 (#930)
49ba008 Support PL 1.8 (#910)
78cbcdd Bump lightning-utilities from 0.3.0 to 0.4.1 in /requirements (#927)
c91726f minor dcgan-import fix (#921)
e9c70c6 skipif for future jsonargparse (#926)
d1c05c4 CI: Use new syntax for setting github output (#924)
cbe220c Comment out kwargs in retinanet until jsonargparse fix (#925)
11d2283 Avoid container name conflict in GPU CI (#923)
8f06ddf CI: switch to Lit pool (#913)
1cbcb93 CI: unify CPU tests (#915)
f38bf89 Add otaj to the CI/CD codeowners (#920)
f347c81 drop cite (#917)
4184b38 update dependabot (#914)
5f9d836 We have new Core Contributors! (#911)
c835f58 drop update.sh (#916)
25c82f5 use lit utilities (#907)
9e1ebb3 docs: install package (#909)
afc8ea3 clean links in changelog (#908)
d108329 swav-improvements (#903)
7f8ede8 Revision of dataset normalizations (#898)
a8bbfc9 Revision of SimSiam module and tests (#891)
79c6e24 Revision `datasets.kitti_dataset.KittiDataset` (#896)
9dd72c4 [pre-commit.ci] pre-commit suggestions (#895)
9d757d6 CVE-2007-4559 Patch (#894)
d8ff64f Revision of BYOL module and tests (#874)
6f58d71 Metrics (#887)
ac98469 Revision of MNIST module (#873)
cbe4143 Dataset specific args method to CIFAR10, ImageNet, MNIST, and STL10 (#890)
e97ae97 Bump python version used by pre-commit (#888)
051a15b fixed header text (#885)
84fc6b8 Bump actions/checkout from 2 to 3 (#884)
d1c0959 CI: Update azure pipelines config path (#869)
8fb0869 revisit Metrics (#878)
eaf6322 fix URL on windows (#882)
665f7be Revision models.vision.unet, models.vision.segmentation (#880)
2a1f134 Revision of SimCLR transforms (#857)
9f3dd69 FashionMNIST/EMNIST Datamodules (#871)
a4855d0 CI: Reuse check schema (#868)
70ce46b Bump actions/upload-artifact from 2 to 3 (#879)
7abe88a CI: Update docker images (#854)
bcbbf6a Feature - `ArrayDataset` (#872)
9619d5f BYOL weight update callback (#867)
7d2a9a1 Binary MNIST/EMNIST Datasets and Datamodules (#866)
645a66f limit mistune version to unblock doctest (#870)
86f5d09 (bugfix/rich-auto-refresh) Added tests, updated doc-strings for Dummy Datasets (#865)
709b2c9 [CI] Make mypy run with installed dev environment (#864)
7e8b746 CI: Update Windows from 2019 to 2022 (#860)
e7458ad Bump actions/cache from 2 to 3 (#863)
e1479d3 Bump codecov/codecov-action from 1 to 3 (#862)
7b5baf9 Bump actions/setup-python from 2 to 4 (#861)
693fefc CI: Add dependabot for GitHub Actions (#855)
cbad552 Fix tests for PL 1.7.0 and TV 0.13 (#859)
d90f8b4 CI: Update `ready` label (#856)
5481b6e Followup of #843 - adding docstring (#847)
f3b17aa test readme (#845)
acc0c98 Reviewing GAN basics, VisionDataModule, MNISTDataModule, CIFAR10DataModule (#843)
675b176 Introduce catch_warnings fixture (#844)
ac3d56d docs: fix coverage badge
388c47f fix CI and update docs (#842)
8aefe66 Remove windows tests (#841)
f88f7f8 Remove warning from __init__ docstrings (#838)
42e7e91 Update MacOS CI (#840)
cb4bcc2 Rename flag to_review -> under_review (#837)
cf670cd `to_review` flag (#835)
dd4863f Update CODEOWNER (#836)
58ee725 Fix/links and docs (#829)
7117399 Install black in development requirements (#834)
da809bb [pre-commit.ci] pre-commit suggestions (#831)
52e4c50 fix-typo-mnist-transforms-in-other-datamodules (#828)
58cb612 Add back the deprecated documentation and fix CI (#827)
5d61197 Update self_supervised.rst (#823)
1e962eb Updated SparseML callback for latest PyTorch Lightning (#822)
9711eef Updated torch version to 1.10.X (#815)
55cb279 support number of channels!=3 in YOLOConfiguration (#806)
72614f0 Removing extra flatten (#809)
3842cf3 [pre-commit.ci] pre-commit suggestions (#820)
142cfb4 update slack link (#821)
2415b49 [pre-commit.ci] pre-commit suggestions (#792)
22e494e rolling 0.6dev
ba899e8 (tag: 0.5.0) Prepare v0.5.0 (#787)
```

</details>

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does **only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes?
- [x] Did you write any **new necessary tests**? \[not needed for typos/docs\]
- [x] Did you verify new and **existing tests** pass locally with your changes?
- [x] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## PR review

- [x] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by otaj** [comment]
Thank you Aki! Can I have one small request - not have this as `0.6.0`, but more as `0.5.1`.

The reason is simple, since master is currently producing huge amount of warnings (intentionally, but still a lot), I would rather not push out "whole" minor version.

Any opinions @Borda?

**Comment by akihironitta** [comment]
Thank you @otaj! That's one thing I wanted to confirm with you and @Borda here in this PR :) cc @Borda

**Review by Borda — CHANGES_REQUESTED** [review]
Is this the only change, no fixes from last December?
In such case we shall be rolling 0.5.1 :thinking:

File: pl_bolts/__about__.py [hunk 1]
```diff
@@ -1,9 +1,11 @@
...
+
+__version__ = "0.6.0"
+__author__ = "Lightning AI et al."
+__author_email__ = "pytorch@lightning.ai"
 __license__ = "Apache-2.0"
-__copyright__ = f"Copyright (c) 2020-2021, {__author__}"
-__homepage__ = "https://github.com/PyTorchLightning/lightning-bolts"
+__copyright__ = f"Copyright (c) 2020-{time.strftime('%Y')}, {__author__}."
+__homepage__ = "https://github.com/Lightning-AI/lightning-bolts"
 __docs__ = "PyTorch Lightning Bolts is a community contribution for ML researchers."
```
Comment (rohitgr7) [inline pl_bolts/__about__.py:9]:
```suggestion
__docs__ = "Lightning Bolts is a community contribution for ML researchers."
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -4,22 +4,43 @@ All notable changes to this project will be documented in this file.
...
-
-### Deprecated
-
-
-### Removed
-
+- Updated SparseML callback for latest PyTorch Lightning ([#822](https://github.com/Lightning-AI/lightning-bolts/822))
+- Updated torch version to 1.10.X ([#815](https://github.com/Lightning-AI/lightning-bolts/815))
+- Dataset specific args method to CIFAR10, ImageNet, MNIST, and STL10 ([#890](https://github.com/Lightning-AI/lightning-bolts/890))
+- use lit utilities ([#907](https://github.com/Lightning-AI/lightning-bolts/907))
```
Comment (Borda) [inline CHANGELOG.md:14]:
```suggestion
- migrate to use `lightning-utilities` ([#907](https://github.com/Lightning-AI/lightning-bolts/907))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -4,22 +4,43 @@ All notable changes to this project will be documented in this file.
...
+- Support PL 1.8 ([#910](https://github.com/Lightning-AI/lightning-bolts/910))
```
Comment (Borda) [inline CHANGELOG.md:15]:
```suggestion
```

Comment (Borda) [inline CHANGELOG.md:15]:
```suggestion
- Support PyTorch Lightning v1.8 ([#910](https://github.com/Lightning-AI/lightning-bolts/910))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -4,22 +4,43 @@ All notable changes to this project will be documented in this file.
...
+
+- Major revision of Bolts
+    * `to_review` flag ([#835](https://github.com/Lightning-AI/lightning-bolts/pull/835))
+    * Rename flag to_review -> under_review ([#837](https://github.com/Lightning-AI/lightning-bolts/pull/837))
+    * Reviewing GAN basics, VisionDataModule, MNISTDataModule, CIFAR10DataModule ([#843](https://github.com/Lightning-AI/lightning-bolts/pull/843))
+    * (bugfix/rich-auto-refresh) Added tests, updated doc-strings for Dummy Datasets ([#865](https://github.com/Lightning-AI/lightning-bolts/pull/865))
+    * Binary MNIST/EMNIST Datasets and Datamodules ([#866](https://github.com/Lightning-AI/lightning-bolts/pull/866))
+    * Feature - `ArrayDataset` ([#872](https://github.com/Lightning-AI/lightning-bolts/pull/872))
```
Comment (Borda) [inline CHANGELOG.md:23]:
```suggestion
    * `ArrayDataset` ([#872](https://github.com/Lightning-AI/lightning-bolts/pull/872))
```

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,22 +4,43 @@ All notable changes to this project will be documented in this file.
...
+    * BYOL weight update callback ([#867](https://github.com/Lightning-AI/lightning-bolts/pull/867))
+    * Revision models.vision.unet, models.vision.segmentation ([#880](https://github.com/Lightning-AI/lightning-bolts/pull/880))
+    * Revision of SimCLR transforms ([#857](https://github.com/Lightning-AI/lightning-bolts/pull/857))
+    * FashionMNIST/EMNIST Datamodules ([#871](https://github.com/Lightning-AI/lightning-bolts/pull/871))
+    * revisit Metrics ([#878](https://github.com/Lightning-AI/lightning-bolts/pull/878))
+    * Revision of BYOL module and tests ([#874](https://github.com/Lightning-AI/lightning-bolts/pull/874))
+    * Metrics ([#887](https://github.com/Lightning-AI/lightning-bolts/pull/887))
```
Comment (Borda) [inline CHANGELOG.md:30]:
```suggestion
    * revisit Metrics ([#887](https://github.com/Lightning-AI/lightning-bolts/pull/887), [#878](https://github.com/Lightning-AI/lightning-bolts/pull/878))
    * Revision of BYOL module and tests ([#874](https://github.com/Lightning-AI/lightning-bolts/pull/874))
```

**Review by Borda — APPROVED** [review]

**Review by rohitgr7 — APPROVED** [review]

**Pull request merged**